### PR TITLE
fix(api): rps rest api's are updated

### PR DESCRIPTION
**Pull request by madhavilosetty-intel**

### CODE REVIEW

File: src/reactjs/components/CIRAEditor/CiraEditor.tsx [hunk 1]
```diff
@@ -74,18 +74,19 @@ CiraEditorState
...
       this.setState({
         showMessage: true,
-        message: response,
+        message: `CIRA Config ${String(this.state.selectedCiraConfigs[0].configName)} deleted`,
         type: 'success',
         updateCiraGrid: !isFalsy(this.state.updateCiraGrid),
         selectedCiraConfigs: ''
       })
     } else {
+      const message = response.data.message != null ? response.data.message : response.data.error
```
Comment (rsdmike) [inline src/reactjs/components/CIRAEditor/CiraEditor.tsx:86]: can use `??`

File: src/reactjs/components/CIRAEditor/CiraEditor.tsx [hunk 2]
```diff
@@ -74,18 +74,19 @@ CiraEditorState
     const configName = encodeSpecialCharacters(this.state.selectedCiraConfigs[0].configName)
     const server: string = this.props.rpsServer != null ? this.props.rpsServer : ''
     const response = await HttpClient.delete(`${server}/api/v1/admin/ciraconfigs/${configName}`, rpsKey)
-    if (response === `CIRA Config ${String(this.state.selectedCiraConfigs[0].configName)} successfully deleted`) {
+    if (response.status === 204) {
```
Comment (rsdmike) [inline src/reactjs/components/CIRAEditor/CiraEditor.tsx:77]: ❤️

File: src/reactjs/components/ProfileEditor/ProfileActionFlyout.tsx [hunk 3]
```diff
@@ -272,42 +272,63 @@ profileFlyoutState
...
       ciraConfigName: this.state.profileFormDetails.networkConfigName !== 'dhcp_disabled'
         ? this.state.profileFormDetails.ciraConfigName
         : ''
     }
-    const payload = this.removeUnAssignedProperties(this.removeMEBxFieldsforCCM(obj))
+    let payload = this.removeUnAssignedProperties(this.removeMEBxFieldsforCCM(obj))
     const server: string = this.props.rpsServer != null ? this.props.rpsServer : ''
+    const cleanObj = (payload): any => {
+      for (const propName in payload) {
+        console.log('propName:', propName)
```
Comment (rsdmike) [inline src/reactjs/components/ProfileEditor/ProfileActionFlyout.tsx:289]: do we need to keep the console.logs? I think we should remove them all

File: src/reactjs/components/CIRAEditor/CiraEditor.tsx (hunk 1 above)
Comment (madhavilosetty-intel) [reply src/reactjs/components/CIRAEditor/CiraEditor.tsx:86]: done

File: src/reactjs/components/ProfileEditor/ProfileActionFlyout.tsx (hunk 3 above)
Comment (madhavilosetty-intel) [reply src/reactjs/components/ProfileEditor/ProfileActionFlyout.tsx:289]: done

File: src/reactjs/components/services/HttpClient.ts [hunk 4]
```diff
@@ -62,6 +62,11 @@ export class HttpClient {
 
   // Handles the response from the server as text
   static async fetchRPS (url, options = {}): Promise<any> {
```
[the remark this answers is not in the document]
Comment (madhavilosetty-intel) [reply src/reactjs/components/services/HttpClient.ts:64]: Yes, response.json() doesn't have the status code info. So, returning an object with status code and json response.

File: src/reactjs/components/CIRAEditor/CiraEditor.tsx (hunk 2 above)
Comment (madhavilosetty-intel) [reply src/reactjs/components/CIRAEditor/CiraEditor.tsx:77]: 👍

**Review by rsdmike — APPROVED** [review]

**Pull request merged**